client/keys/add.go

```diff
@@ -94,18 +103,28 @@ func runAddCmd(cmd *cobra.Command, args []string) error {
 		}
 
 		// ask for a password when generating a local key
-		if !viper.GetBool(client.FlagUseLedger) {
-			encryptPassword, err = client.GetCheckPassword(
-				"Enter a passphrase to encrypt your key to disk:",
-				"Repeat the passphrase:", buf)
-			if err != nil {
-				return err
+
+		if !pubKeyFlag.Changed {
+			if !viper.GetBool(client.FlagUseLedger) {
+				encryptPassword, err = client.GetCheckPassword(
+					"Enter a passphrase to encrypt your key to disk:",
+					"Repeat the passphrase:", buf)
+				if err != nil {
+					return err
+				}
 			}
 		}
 	}
 
-	interactive := viper.GetBool(flagInteractive)
```

Comment:
This var declaration should be moved closer to where it's first used, i.e. downward rather than upward. Not a big deal either way though

Reply:
Done!